zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -718,24 +718,29 @@ where
     }
 
     for change_value in proposal.balance().proposed_change() {
-        match change_value {
-            ChangeValue::Sapling { value, memo } => {
-                let memo = memo.as_ref().map_or_else(MemoBytes::empty, |m| m.clone());
+        let memo = change_value
+            .memo()
+            .map_or_else(MemoBytes::empty, |m| m.clone());
+        match change_value.output_pool() {
+            ShieldedProtocol::Sapling => {
                 builder.add_sapling_output(
                     internal_ovk(),
                     dfvk.change_address().1,
-                    *value,
+                    change_value.value(),
                     memo.clone(),
                 )?;
                 sapling_output_meta.push((
                     Recipient::InternalAccount(
                         account,
                         PoolType::Shielded(ShieldedProtocol::Sapling),
                     ),
-                    *value,
+                    change_value.value(),
                     Some(memo),
                 ))
             }
+            ShieldedProtocol::Orchard => {
+                unimplemented!("FIXME: implement Orchard change output creation.")
```

Comment:
```suggestion
                #[cfg(feature = "orchard")]
                unimplemented!("FIXME: implement Orchard change output creation.")
```

Also, the case where Orchard change is proposed but the `orchard` feature flag is not enabled needs to be handled here.